html-report/src/components/ColorizedLogsComponent.js

```diff
@@ -56,58 +57,116 @@ export default class ColorizedLogsComponent extends Component {
-                                            return "line verbose";
-                                        }
-                                    }
-                                }
-
-                                return (<tr className={selectStyle(level)}>
-                                    <td>
-                                        {process}
-                                    </td>
-                                    <td>
-                                        {tag}
-                                    </td>
-                                    <td>{level}</td>
-                                    <td className="formatted-time">{time}</td>
-                                    <td>{message}</td>
-                                </tr>);
-                            }
-                        )}
+                        {!!this.state.logs && this.state.logs
+                            .map((message, index) =>
+                                     (<tr className={_selectStyle(message.priority)} key={index}>
+                                         <td>{message.processId}</td>
+                                         <td>{message.tag}</td>
+                                         <td>{message.priority}</td>
+                                         <td className="formatted-time">{message.time}</td>
+                                         <td>
+                                             <pre>{message.message}</pre>
+                                         </td>
+                                     </tr>)
+                            )}
                         </tbody>
                     </table>
                     {this.state.logs == null && <ReactLoading className="center"
                                                               type="bubbles"
                                                               color="#ff0000"
-                                                              delay="1"/>}
+                                                              delay={1}/>}
                 </div>
             </div>
         );
     }
-}
\ No newline at end of file
+}
+
+function _selectStyle(logLevel) {
+    switch (logLevel) {
+        case "W":
+            return "line warn";
+        case "D": {
+            return "line debug";
+        }
+        case "E": {
+            return "line error";
+        }
+        case "I": {
+            return "line info";
+        }
+        case "A": {
+            return "line assert";
+        }
+        case "V": {
+            return "line verbose";
+        }
+    }
+}
+
+/**
+ * Pattern for header (MM-DD HH:MM:SS.mmm PID-TID/AppName LEVEL/TAG: message). Example:
+ *
+ * <pre>04-06 16:21:43.746 1852-25383/? D/GraphicBufferSource: got buffer with new dataSpace #104</pre>
+ *
+ * Group 1: Date + Time
+ * Group 2: PID
+ * Group 3: TID (hex on some systems!)
+ * Group 4: Log Level character
+ * Group 5: Tag
+ */
+function _parseLogcatMessages(response) {
```

Comment:
updated parsing so it supports messages with one header and multiline message body

Reply:
![image](https://user-images.githubusercontent.com/725526/78584412-17b93b00-7830-11ea-8326-b4efeb743b7a.png)